### ISSUE TITLE
Alpha Complex Filtraton

### ISSUE THREAD

**Issue by arksch** [scikit-tda/cechmate#10]
Filtration values of the simplices in the Alpha complex can be much larger than they should be. I found this issue when experimenting with Alpha complexes built on points from a uniform normal distribution in 2D.
To reproduce this problem, you can call the following code a few times and see that there will be filtration values much larger than 10, which is basically impossible for the given distribution.
I also double checked that the referenced triangles actually have a much smaller circumradius than what is given in the Alpha complex. It seems to happen for those triangles that already have quite a large filtration value.

```python
import numpy as np
from cechmate.filtrations import Alpha
points = np.random.randn(20, 2)
simplices, filtration = zip(*Alpha().build(points))
max(filtration)
```

Here an example data set that would gives a maximum filtration value of 11.18, namely for the triangle (1, 4, 10) and the line (4, 10). The line refers to points (-0.66838721, -2.21357309) and (-0.68879532,  1.36314908), so the actual filtration value should be 1.800.

```
points = np.array(
      [[ 0.01743489,  0.83907818],
       [-0.57518843,  0.46536324],
       [-0.19659281, -0.66731467],
       [ 1.52911009, -0.68218385],
       [-0.66838721, -2.21357309],
       [ 1.14180137,  0.79701124],
       [-0.05349503, -2.25566765],
       [-0.27223817,  0.77621451],
       [ 0.38597224,  1.15861246],
       [-0.29454972,  1.71746955],
       [-0.68879532,  1.36314908],
       [-0.47834989, -1.57854915],
       [ 0.94477495, -0.38586968],
       [-0.04377718, -0.84981483],
       [-0.03082609, -1.63861901],
       [ 1.73579262, -0.02458939],
       [ 0.50910058,  0.66446628],
       [ 1.88017434,  1.66114513],
       [ 1.47186944, -0.68486166]])
```

**Comment by arksch** [comment]
As this is important to my research, I will investigate this issue and might just send you a pull request in the coming days.

**Comment by ctralie** [comment]
Thank you so much for bringing this to my attention!  Yes, if you could
have a look and make a pull request, that would be a huge contribution.  If
you are on a tighter timescale, note that GUDHI also has an alpha
filtration module:
http://gudhi.gforge.inria.fr/doc/latest/group__alpha__complex.html
GUDHI is harder to install, which is part of what spurred me to write this
library, but it should work well.

Regardless, I'll have a look at this as soon as I am able.
Many thanks,
Chris

**Comment by arksch** [comment]
Hi Chris,
in the example I posted, Gudhi even creates filtration values of over 100. Quite odd. Maybe an issue of the miniball algorithm.
It also seems that this error does not affect the actual persistence module, as in all instances I found there are a 2-simplex and one of its faces with the same value, so it will only create ephemeral homology classes of zero persistence.
I forked and keep you posted.
